app/controllers/vm_cloud_controller.rb

```diff
@@ -309,6 +307,92 @@ def cancel_action(message)
     replace_right_cell
   end
 
+  def evacuate
+    assert_privileges("instance_evacuate")
+    @record = find_by_id_filtered(VmOrTemplate, params[:id]) # Set the VM object
+    if @record.is_available?(:evacuate) && !@record.ext_management_system.nil?
+      drop_breadcrumb(
+        :name => _("Evacuate Instance '%{name}'") % {:name => @record.name},
+        :url  => "/vm_cloud/evacuate"
+      ) unless @explorer
+      @in_a_form = true
+      @refresh_partial = "vm_common/evacuate"
+    else
+      add_flash(_("Unable to evacuate %{instance} \"%{name}\": %{details}") % {
+        :instance => ui_lookup(:table => 'vm_cloud'),
+        :name     => @record.name,
+        :details  => @record.is_available_now_error_message(:evacuate)}, :error)
+    end
+  end
+  alias_method :instance_evacuate, :evacuate
+
+  def evacuate_form_fields
+    assert_privileges("instance_evacuate")
+    @record = find_by_id_filtered(VmOrTemplate, params[:id])
+    host_ems = @record.try!(:host).try!(:ext_management_system)
+    clusters = []
+    hosts = []
+    if host_ems
+      clusters = host_ems.ems_clusters.map do |c|
+        {:id => c.id, :name => c.name}
+      end
+      hosts = host_ems.hosts.map do |h|
```

Comment:
We need to filter out hosts that don't have an ems_cluster since it causes a nil-related exception. Replace line 339 with something like this:
hosts = host_ems.hosts.select{|h|!h.ems_cluster.nil?}.map do |h|